queueUI/mainwindow.py

```diff
@@ -32,6 +45,47 @@ def __init__(self):
         settings.restoreWindowGeometry(self, "mainwindow_geometry")
         print("Settings are saved in:", settings.fileName())
 
+        self.RM = REManagerAPI()
+        self.REOpenButton.clicked.connect(self.openRunEngine)
+        self.RECloseButton.clicked.connect(self.closeRunEngine)
+
+    # RE Function:
+    def openRunEngine(self):
+        """Checks the status of the RE. Then opens the RE."""
+        if not self.RM.status().get("worker_environment_exists"):
+            # BUG: Status bar in app window does not print the statement below
+            self.setStatus("RE Environment is opening.", timeout=0)
+            self.RM.environment_open()
+            self.RM.wait_for_idle()
+            self.REEnvironmentStatusLabel.setText("Open")
+            self.REOpenButton.setEnabled(False)
+            self.RECloseButton.setEnabled(True)
+            self.setStatus("RE Environment is now open.", timeout=0)
+
+            # print(f"status = {self.RM.status().get('worker_environment_exists')}")
+        else:
+            pass
+
+    def closeRunEngine(self):
+        """Checks the status of the RE. Then closes the RunEngine."""
+        if self.RM.status().get("worker_environment_exists"):
+            self.setStatus("RE Environment is closing.", timeout=0)
+            self.RM.environment_close()
+            self.RM.wait_for_idle()
+            self.REEnvironmentStatusLabel.setText("Closed")
+            self.RECloseButton.setEnabled(False)
+            self.REOpenButton.setEnabled(True)
+            self.setStatus("RE Environment is now closed.", timeout=0)
```

Comment:
This code is very similar to lines 59-63. Is there a better way to use this code block so when we need to change it we don't change it in 2 place?